sphinx/screens-detail/tribe-members-list/tribe-members-list/src/main/java/chat/sphinx/tribe_members_list/ui/adapter/TribeMembersListAdapter.kt

```diff
@@ -0,0 +1,310 @@
+            tribeMemberHolderViewState = tribeMembers.getOrNull(position) ?: let {
+                tribeMemberHolderViewState = null
+                return
+            }
+            disposable?.dispose()
+
+            tribeMemberHolderViewState?.apply {
+                when (this) {
+                    is TribeMemberHolderViewState.Loader -> {
+                        bindLoader(binding)
+                    }
+                    is TribeMemberHolderViewState.Member -> {
+                        if (contactDto != null) {
+                            bindContactDetails(binding, contactDto, showInitial)
+                        }
+                    }
+                    is TribeMemberHolderViewState.Pending -> {
+                        if (contactDto != null) {
+                            bindContactDetails(binding, contactDto, showInitial)
+                            bindAdminFunctions(binding, contactDto, position)
+                        }
+                    }
+                    is TribeMemberHolderViewState.PendingTribeMemberHeader -> {
+                        bindHeader(binding, binding.root.context.getString(R.string.pending_tribe_members_header))
+                    }
+                    is TribeMemberHolderViewState.TribeMemberHeader -> {
+                        bindHeader(binding, binding.root.context.getString(R.string.tribe_members_list_header))
+                    }
+                }
+            }
+        }
+
+        private fun bindLoader(binding: LayoutTribeMemberHolderBinding) {
+            binding.includeLoadingMoreMembers.root.goneIfFalse(true)
+        }
+
+        private fun bindContactDetails(
+            binding: LayoutTribeMemberHolderBinding,
+            contactDto: ContactDto,
+            shouldShowInitial: Boolean
+        ) {
+            binding.apply {
+                layoutConstraintTribeMemberContainer.visible
+                includeLoadingMoreMembers.root.gone
+                layoutConstraintTribeMemberHeaderContainer.gone
+                constraintLayoutTribeMemberRequestActions.gone
+
+                textViewMemberName.text = contactDto.alias
+
+                textViewMemberFirstInitial.text = contactDto.alias?.first().toString()
+                textViewMemberFirstInitial.goneIfFalse(shouldShowInitial)
+
+                textViewMemberInitials.text = contactDto.alias?.getInitials() ?: ""
+                textViewMemberInitials.setBackgroundRandomColor(chat.sphinx.resources.R.drawable.chat_initials_circle)
+
+                if (contactDto.photo_url?.isNotEmpty() == true) {
+                    imageViewMemberPicture.visible
+
+                    onStopSupervisor.scope.launch(viewModel.mainImmediate) {
+                        imageLoader.load(
+                            imageViewMemberPicture,
+                            contactDto.photo_url!!,
+                            imageLoaderOptions
+                        )
+                    }
```

Comment:
both the job and the imageLoader's disposable need to be added to the ViewHolder's list such that they can be cancelled upon re-bind.